Bug: knowledge search_components tests fail with ModuleNotFoundError: models.task_context

## Problem

All 4 tests in `tests/knowledge/search_components/` fail during collection with:

```
knowledge/search_components/__init__.py:21: in <module>
    from .analytics import SearchAnalytics, get_analytics
knowledge/search_components/analytics.py:15: in <module>
    from models.task_context import SearchAnalyticsContext
E   ModuleNotFoundError: No module named 'models.task_context'
```

The `knowledge/search_components/analytics.py` imports `SearchAnalyticsContext` from `models.task_context`, but that module doesn't exist (or isn't on PYTHONPATH when running tests from the project root).

Affected tests:
- `bm25_test.py`
- `context_tracker_test.py`
- `query_classifier_test.py`
- `reranking_blend_test.py`

## Reproduction

```bash
python3 -m pytest autobot-backend/tests/knowledge/search_components/ -v
```

## Discovered During

Working on #2037 (moving co-located tests to tests/ directory). The failure is pre-existing — it occurred both before and after the file move.

## Impact

**Severity: medium** — 4 tests are permanently broken and provide no coverage for the search components (BM25, reranking, query classifier, context tracker).

--- a/autobot-backend/tests/knowledge/__init__.py
+++ b/autobot-backend/tests/knowledge/__init__.py
@@ -1,0 +1,3 @@
+# AutoBot - AI-Powered Automation Platform
+# Copyright (c) 2025 mrveiss
+# Author: mrveiss

--- a/autobot-backend/tests/knowledge/search_components/__init__.py
+++ b/autobot-backend/tests/knowledge/search_components/__init__.py
@@ -1,0 +1,3 @@
+# AutoBot - AI-Powered Automation Platform
+# Copyright (c) 2025 mrveiss
+# Author: mrveiss

